### PR TITLE
Drop down menu + Avabilities

### PR DESCRIPTION
**Pull request by BlobMaster41**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -137,4 +137,4 @@ dist
 .idea
 .idea/*
 
-emulatorSave
\ No newline at end of file
+#emulatorSave
```
Comment (RominMartignat) [inline .gitignore:140]: tu dois faire passer les tests

File: src/src/engine/components/NavigationBar.tsx [hunk 2]
```diff
@@ -51,6 +52,7 @@ export class NavigationBar extends React.Component<unknown, NavigationBarState>
      * This function returns the links to the admin commands if the user is an admin
      * @returns {JSX.Element[]} The links to the admin commands
      * @private
+     *
```
Comment (DJLawton) [inline src/src/engine/components/NavigationBar.tsx:55]: why

File: src/src/engine/components/ComponentAvailabilities.tsx [hunk 3]
```diff
@@ -41,190 +38,139 @@ export class ComponentAvailabilities extends Component<ComponentAvailabilitiesPr
...
-    }
+        /*const previousWeekStart = DayPilot.Date.today().firstDayOfWeek().addDays(-7);
+        const previousWeekEnd = previousWeekStart.addDays(7);
 
-    private changeColorToGray = (args: DayPilotArgumentTimeRange): void => {
+        if (DayPilot.Util.overlaps(args.cell.start, args.cell.end, previousWeekStart, previousWeekEnd)) {
+            args.cell.properties.disabled = true;
+            args.cell.properties.backColor = "#eeeeee";
+        }*/
+    }
```
Comment (DJLawton) [inline src/src/engine/components/ComponentAvailabilities.tsx:137]: Remove commented code

File: src/src/engine/utils/DateManager.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        let newDate = new Date(numberOfMs + msToAdd);
+        return this.changeDateToDayPilotFormat(newDate);
+    }
+
+    public changeDateToDayPilotFormat(date: Date) {
+        return date.toISOString().slice(0, -5);
+    }
+}
+
+export const ManagerDate = new DateManager();
```
Comment (DJLawton) [inline src/src/engine/utils/DateManager.ts:50]: Should be named DateManager not ManagerDate

File: src/src/engine/pages/Availabilities.tsx [hunk 5]
```diff
@@ -1,39 +1,231 @@
...
+                            listOfUnavailbility.push(eventToPush);
+                        }
+                    }
+                }
+            }
+        }
+
+        //this.state.timesUnavailable = listOfUnavailbility;
+
+        return listOfUnavailbility;
```
Comment (DJLawton) [inline src/src/engine/pages/Availabilities.tsx:228]: Comment or simplify

File: src/src/engine/pages/Availabilities.tsx [hunk 6]
```diff
@@ -1,39 +1,231 @@
...
+                            if(eventStart.getTime() <= startDateTime.getTime() && endDateTime.getTime() <= eventEnd.getTime()) {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return false;
```
Comment (DJLawton) [inline src/src/engine/pages/Availabilities.tsx:183]: comment of simplify

File: src/src/engine/pages/Availabilities.tsx [hunk 7]
```diff
@@ -1,39 +1,231 @@
...
+        date.setDate(date.getDate() + day);
+        date.setHours(numberOfMinutes / 60);
+        date.setMinutes(numberOfMinutes % 60);
+
+        return date;
+    }
+
+    private toUTC(date: Date): DayPilot.Date {
+        return new DayPilot.Date(date, true);
+    }
```
Comment (DJLawton) [inline src/src/engine/pages/Availabilities.tsx:125]: Should be in DateManager

File: src/src/engine/pages/Availabilities.tsx [hunk 8]
```diff
@@ -1,39 +1,231 @@
...
+    popupActive: boolean;
+
+    currentWeekStart: Date;
+    currentWeekEnd: Date;
+    selectedDate: Date;
+}
+
+let curr = new Date;
+let firstday = new Date ((new Date(curr.setDate(curr.getDate() - curr.getDay()))).setHours(0,0,0,0));
+let lastday = new Date ((new Date(curr.setDate(curr.getDate() - curr.getDay() + 6))).setHours(0,0,0,0));
```
Comment (DJLawton) [inline src/src/engine/pages/Availabilities.tsx:25]: Comment or clarify

**Review by RominMartignat — APPROVED** [review]
good pour le reste

**Review by VillyGH — APPROVED** [review]
cool code

**Pull request merged**